pkg/virtctl/vmexport/vmexport.go

```diff
@@ -0,0 +1,639 @@
+	}
+
+	fmt.Printf("VirtualMachineExport '%s/%s' deleted succesfully\n", namespace, vmexportName)
+
+	// If it exists, attempt to delete the related secret
+	if err := deleteExportSecret(client, namespace); err != nil {
+		return err
+	}
+
+	return nil
+}
+
+// downloadVirtualMachineExport handles the process of downloading the requested volume from a VirtualMachineExport object
+func downloadVirtualMachineExport(client kubecli.KubevirtClient, namespace string) error {
+	if shouldCreate {
+		if err := createVirtualMachineExport(client, namespace); err != nil {
+			return err
+		}
+	}
+
+	vmexport, err := getVirtualMachineExport(client, namespace)
+	if err != nil {
+		return err
+	}
+	if vmexport == nil {
+		return fmt.Errorf("Unable to get '%s/%s' VirtualMachineExport", namespace, vmexportName)
+	}
+	// Wait for the vmexport object to be ready
+	if err := waitForVirtualMachineExport(client, namespace, processingWaitInterval, processingWaitTotal); err != nil {
+		return err
+	}
+	// Download the exported volume
+	if err := downloadVolume(client, vmexport); err != nil {
+		return err
+	}
+
+	fmt.Println("Cleaning resources...")
+	if err := deleteVirtualMachineExport(client, namespace); err != nil {
+		return err
+	}
+
+	return nil
+}
+
+// downloadVolume handles the process of downloading the requested volume from a VirtualMachineExport
+func downloadVolume(client kubecli.KubevirtClient, vmexport *exportv1.VirtualMachineExport) error {
+	output, err := os.Create(outputFile)
+	if err != nil {
+		return err
+	}
+	defer util.CloseIOAndCheckErr(output, nil)
+
+	// Extract the URL from the vmexport
+	downloadUrl, httpHeader, err := getUrlFromVirtualMachineExport(client, vmexport)
+	if err != nil {
+		return err
+	}
+
+	resp, err := handleHTTPRequest(client, vmexport, downloadUrl, httpHeader)
+	if err != nil {
+		return err
+	}
+	defer resp.Body.Close()
+
+	// Check server response
+	if resp.StatusCode != http.StatusOK {
+		return fmt.Errorf("Bad status: %s", resp.Status)
+	}
+
+	// Lastly, copy the file to the expected output
+	if err := copyFileWithProgressBar(output, resp); err != nil {
+		return err
+	}
+
+	fmt.Println("Download finished succesfully")
+	return nil
+}
+
+// getUrlFromVirtualMachineExport inspects the VirtualMachineExport status to fetch the extected URL
+func getUrlFromVirtualMachineExport(client kubecli.KubevirtClient, vmexport *exportv1.VirtualMachineExport) (string, string, error) {
+	var downloadUrl string
+	var httpHeader string
+
+	if vmexport.Status.Links.External.Volumes == nil {
+		return "", "", fmt.Errorf("Unable to access the volume info from '%s/%s' VirtualMachineExport", vmexport.Namespace, vmexport.Name)
+	}
+
+	for _, exportVolume := range vmexport.Status.Links.External.Volumes {
+		// Access the requested volume
+		if exportVolume.Name == volumeName {
+			for _, format := range exportVolume.Formats {
+				// If the --raw flag is used, we download the uncompressed file
+				if downloadRaw {
```

Comment:
So raw doesn't make much sense in the context of archives. So we have two distinct sets formats
1. KubeVirt content type ->
  a. Raw
  b. gzip
2. Archive content type ->
  a. Dir
  b. tar.gz

Dir will return a list of files, which doesn't make much sense in the context of virtctl download. So I would ignore --raw if you detect archive content type and just download the tar.gz file.

Reply:
Okay, however, what would be the best way to handle Dir downloads? Should I just skip them?

Reply:
Just skip them for now I would say.